NMS-14210: Add documentation for SELinux as a requirement to run OpenNMS (#5502)
OpenNMS/opennms: docs/modules/deployment/pages/core/centos-rhel8/selinux.adoc

If you have SELinux enabled on your {page-component-title} server, there must be a policy configured to allow {page-component-title} to bind to the ICMP service.

> To verify that you have SELinux running and enabled, use the `sestatus` command.

Reviewer: ```suggestion
To verify that you have SELinux running and enabled, use the `sestatus` console command.
```

> If enabled, you should see the following:

Reviewer: ```suggestion
If it's enabled, you should see the following:
```

[source, console]
----
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      33
----

> Use `audit2allow -a` to verify if {page-component-title} violates SELinux policies.

Reviewer: ```suggestion
Use `audit2allow -a` to check if {page-component-title} violates SELinux policies.
```

> NOTE: If the `audit2allow` command is missing, run `dnf install policycoreutils-devel` to install package.

Reviewer: ```suggestion
NOTE: If the `audit2allow` command is missing, run `dnf install policycoreutils-devel` to install the package.
```

> Look for entries preventing bind ICMP sockets to bind:

Reviewer: This line doesn't read very well; I'm not 100% sure what it's trying to say.

Author: @indigo423 , could you provide some feedback here?

Reviewer 2: What it tries to say, the audit2allow command will give the user the output which SELinux policies prevent binding ICMP sockets. These policies need to be explicitly allowed from the user.

> Look for entries preventing bind ICMP sockets to bind:

Author: ```suggestion
Look for entries that prevent the binding of ICMP sockets:
```

[source, console]
----
#============= unconfined_service_t ==============
allow unconfined_service_t node_t:icmp_socket node_bind;

============= unconfined_service_t ==============
allow unconfined_service_t port_t:icmp_socket name_bind;
----

Create a type enforcement file (for example, `JniPing.te`):

[source, console]
----
module JniPing 1.0;

require {
        type unconfined_service_t;
        type node_t;
        type port_t;
        class icmp_socket { name_bind node_bind };
}

#============= unconfined_service_t ==============
allow unconfined_service_t node_t:icmp_socket node_bind;
allow unconfined_service_t port_t:icmp_socket name_bind;
----

Create a SElinux module and compile it:

[source, console]
----
checkmodule -M -m -o JniPing.mod JniPing.te
semodule_package -o JniPing.pp -m JniPing.mod
----

Install the SELinux package and enforce it:

[source, console]
----
semodule -i JniPing.pp
getenforce
----

> You can verify the package with the following command: `semodule --list-modules | grep JniPing`.

Reviewer 2: The commands above are based on the `audit2allow` command on a CentOS 8 system.

> You can verify the package with the following command: `semodule --list-modules | grep JniPing`.

Author: ```suggestion
You can verify the package with the following command: `semodule --list-modules | grep JniPing`.

Note that this command is based on the `audit2allow` command on a CentOS 8 system.
```
